test/addons/argocd/start

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import sys
+import json
+
+from drenv import kubectl
+from drenv import minikube
+from drenv import commands
+
+
+VERSION = "release-2.11"
+URL = f"https://raw.githubusercontent.com/argoproj/argo-cd/{VERSION}/manifests/install.yaml"
+
+
+def deploy(cluster):
+    print("Deploying argocd")
+    kubectl.apply("--filename", "argocd-ns.yaml", context=cluster)
+    kubectl.apply("-n", "argocd", "--filename", URL, context=cluster)
+
+
+def wait(cluster):
+    print("Waiting until all deployments are available")
+    kubectl.wait(
+        "deploy",
+        "--all",
+        "--for=condition=Available",
+        "--namespace=argocd",
+        "--timeout=600s",
+        context=cluster,
+    )
+
+
+def get_minikube_profile():
+    out = minikube.profile("list", output="json")
+    profiles = json.loads(out)
+    names = []
+    for profile in profiles["valid"]:
+        names.append(profile["Name"])
+    return names
+
+
+if len(sys.argv) != 2:
+    print(f"Usage: {sys.argv[0]} cluster")
+    sys.exit(1)
+
+os.chdir(os.path.dirname(__file__))
+cluster = sys.argv[1]
+
+deploy(cluster)
+wait(cluster)
+
+# Need switch to argocd ns first, otherwise will hit argocd bug
+# see https://github.com/argoproj/argo-cd/issues/14167
+kubectl.config("use-context", cluster)
+kubectl.config("set-context", "--current", "--namespace=argocd")
```

Comment:
No we cannot do this. All code must work with --context, so we can run code in parallel. There is no concept of current cluster in drenv code.

We also don't change the default namespace, all code must use --namespace to specify the namespace.

If there is no way to workaround the bug, maybe use a temporary kubeconfig file so we don't affect other code running at the same time?

We can export the current kubeconfig to a temporary kubeconfig;

```
kubectl config view --flatten > kubeconfig.tmp
```

Now you can do whatever you want with the kubeconfig.tmp:

```
kubectl.config("use-context", cluster, "--kubeconfig=kubeconfig.tmp")
kubectl.config("set-context", "--current", "--namespace=argocd", "--kubeconfig=kubeconfig.tmp")
```

When we finish with the temporary kubeconfig we can delete it.

The cleanest way to do this is to create a tempoary directory and put the temporary kubeconfig inside it.

```
with tempfile.TempoaryDirectory() as tmp:
    kubeconfig = os.path.join(tmp, "kubeconfig")
    # export current kubeconfig and modify it...
    # run commands with the tempoary kubeconfig...
```

The temporary directory is deleted automatically when we exit the with block. See drenv.cache for example usage.

Reply:
I tested, argocd command right now does not honor context and namespace param, nor for a kubeconfig file. So this try does not work.

Reply:
I tested with argocd command line parameter `--kubeconfig string`, `--kube-context string`, `--context string`, but none of that could help the issue. 
Thanks for finding out that using a KUBECONFIG env entry could work.

Updated code based on your code example.